src/test/datascience/notebook/interrupRestart.vscode.test.ts

```diff
@@ -133,33 +136,54 @@ suite('DataScience - VSCode Notebook - Restart/Interrupt/Cancel/Errors (slow)',
         disposables.push({ dispose: () => showInformationMessage.restore() });
 
         (editorProvider.activeEditor as any).shouldAskForRestart = () => Promise.resolve(false);
+        traceInfo(`Step 4. Before execute`);
         await executeActiveDocument();
+        traceInfo(`Step 5. After execute`);
 
         // Wait for cell to get busy.
         await waitForCondition(async () => assertVSCCellIsRunning(cell), 15_000, 'Cell not being executed');
+        traceInfo(`Step 6. Cell is busy`);
 
         // Wait for ?s, and verify cell is still running.
         assertVSCCellIsRunning(cell);
         // Wait for some output.
         await waitForTextOutputInVSCode(cell, '1', 0, false, 15_000); // Wait for 15 seconds for it to start (possibly kernel is still starting).
+        traceInfo(`Step 7. Cell output`);
 
         // Restart the kernel.
-        let restartPromise = commands.executeCommand('jupyter.notebookeditor.restartkernel');
+        const restartPromise = commands.executeCommand('jupyter.notebookeditor.restartkernel');
 
-        await waitForCondition(async () => assertVSCCellIsNotRunning(cell), 15_000, 'Execution not cancelled');
+        await waitForCondition(
+            async () => {
+                traceInfo(`Step 8 Cell Status = ${cell.metadata.runState}`);
+                return assertVSCCellIsNotRunning(cell);
+            },
+            15_000,
+            'Execution not cancelled first time.'
+        );
 
         // Wait before we execute cells again.
+        traceInfo('Step 9 Wait for restart');
         await restartPromise;
+        traceInfo('Step 10 Restarted');
 
         // Confirm we can execute a cell (using the new kernel session).
         await executeActiveDocument();
+        traceInfo('Step 11 Executed');
 
         // Wait for cell to get busy.
         await waitForCondition(async () => assertVSCCellIsRunning(cell), 15_000, 'Cell not being executed');
+        traceInfo('Step 12 Cells executed after restart');
+        // Wait for some output.
+        await waitForTextOutputInVSCode(cell, '1', 0, false, 15_000); // Wait for 15 seconds for it to start (possibly kernel is still starting).
```

Comment:
This is a possible fix, we wait for output, else we could end up interrupting even before there's any output & resulting in some race condition.